appcelerator/titanium_mobile
# [TIMOB-26484] Android: Added "sound" support to NotificationChannel

**JIRA:**
https://jira.appcelerator.org/browse/TIMOB-26484

**Note:**
This is needed to resolve issue [TIMOB-26432](https://jira.appcelerator.org/browse/TIMOB-26432), where custom sounds attached to individual notifications are ignored on Android 8.0 and newer OS versions. For these OS versions, you must assign the custom sound to a notification channel instead.

**Test:**
1. Create a Titanium classic project with `NotificationSoundTest.js` attached to [TIMOB-26484](https://jira.appcelerator.org/browse/TIMOB-26484).
2. Under your project, create folder: `./platform/android/res/raw`
3. Copy `notification_sound.wav` to the above raw folder.
4. Play this sound file on your desktop so that you'll know what it sounds like later.
5. Build and run this project on an Android 8.0 or newer device.
6. Tap the "Send Notification" button.
7. Verify that you hear the above `notification_sound.wav`. _(This is the fix.)_
8. In the log, verify that you see warning message: `Notification 'sound' property is not supported on Android 8.0 and higher. You must assign sound to its NotificationChannel instead.`
9. Run the same project on an Android 7.1 or older device.
10. Tap the "Send Notification" button.
11. Verify that you hear the `notification_sound.wav`. _(Verifies old method still works.)_
